tools/bisq2hashcat.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python
+
+# This software is Copyright (c) 2017, Dhiru Kholia <dhiru.kholia at gmail.com>
+# and it is hereby released under GPL v2 license.
+#
+# Major parts are borrowed from the "btcrecover" program which is,
+# Copyright (C) 2014-2016 Christopher Gurnee and under GPL v2.
+#
+# See https://github.com/gurnec/btcrecover for details.
+#
+# References,
+#
+# https://github.com/gurnec/btcrecover/blob/master/btcrecover/btcrpass.py
+#
+
+# Updated script in order to work with Bisq wallet v1.9.5 (2022) 
+# Author: 
+#   https://github.com/Banaanhangwagen
+# License: MIT
+
+import os
+import sys
+import base64
+import binascii
+
+# The protobuf parsing code is borrowed from btcrecover, and was
+# initially automatically generated by Google's protoc utility.
+# The precompiled file is shipped in protobuf/wallet_pb2.py with John.
+from protobuf.wallet_pb2 import *
+
+
+def process_file(filename):
+    bname = os.path.basename(filename)
+    try:
+        f = open(filename, "rb")
+        data = f.read()
+    except IOError:
+        e = sys.exc_info()[1]
+        sys.stderr.write("%s\n" % str(e))
+        return
+
+    if "wallet" in bname or b"org.bitcoin.production" in data:
+ #       sys.stderr.write("[WARNING] Cracking .wallet files is a very slow process, try cracking the associated .key file instead!\n")
+        version = 3  # MultiBit Classic .wallet file
+        # def is_wallet_file(wallet_file) from btcrecover
+        wallet_file = open(filename, "rb")
+        wallet_file.seek(0)
+        is_valid_bitcoinj_wallet = False
+        if wallet_file.read(1) == b"\x0a":  # protobuf field number 1 of type length-delimited
```

Comment:
I prefer to use [`with` statement](https://docs.python.org/3/reference/compound_stmts.html#with).

```suggestion
        with open(filename, "rb") as wallet_file:
            wallet_file.seek(0)
            is_valid_bitcoinj_wallet = False
            if wallet_file.read(1) == b"\x0a":  # protobuf field number 1 of type length-delimited
```

See also https://stackoverflow.com/a/31334226.